samples/search_grouding.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DynamicRetrievalMode,
+  GoogleGenerativeAI,
+} from "@google/generative-ai";
+
+async function searchGrounding() {
+  // [START search grounding]
+  const genAI = new GoogleGenerativeAI(process.env.API_KEY);
+  const model = genAI.getGenerativeModel(
+    {
+      model: "gemini-1.5-pro",
```

Comment:
Is there a reason for choosing different models for each sample?

Reply:
I see the two models are now flash and flash-002, is there a reason they are different?

Reply:
Because the logprob is only supported for 002 model. A bit background: Labs are calling vertex for all 002 models. And logprob is a feature only supported by vertex for now.